### PR TITLE
Improve accessibility of the result page

### PR DESCRIPTION
**Pull request by blacksponge**
## Purpose

Improve accessibility of the result page

## Context

\-

## Changes

The page looks and behave mostly the same way as before.

Notable changes:
* Wrap heading content into a button element that is itself in a heading element (`<h3><button>content</button></h3>`) for modules and testcases. Before that a heading with button role was used, as a result the element was loosing the semantic of the heading, this might have impacted people navigating by heading (eg. when using a braille or screen reader).
* Side effect of the change: improve keyboard navigation, the space key can now be used as well as the enter key to toggle module and test cases.

## How to test this PR

1. Navigate to a test result.
2. Open and close module and test case sections using the mouse.
3. Try doing the same using the keyboard (use tab to focus) using both the enter and space key.
4. Verify that no visual regression has appeared.

### CODE REVIEW

**Review by matsduf — COMMENTED** [review]
It looks fine and I see no issues when I test, but Travis fails.

**Comment by blacksponge** [comment]
The issue with travis are probably coming from the change in the html layout. I was waiting for some approval before fixing those.

**Comment by blacksponge** [comment]
E2E tests has been fixed.

**Review by matsduf — APPROVED** [review]

**Comment by pnax** [comment]
### v2023.1 release testing

* I've tested the PR following the instructions in the description and couldn't see any error
* I've used the browser's accessibility tool to look for any accessibility issue and couldn't find any related to the result component.

**Pull request merged**
